src/main/java/org/onecx/search/config/rs/v1/mapper/ExceptionMapper.java

```diff
@@ -0,0 +1,83 @@
+package org.onecx.search.config.rs.v1.mapper;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import jakarta.persistence.OptimisticLockException;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.ConstraintViolationException;
+import jakarta.validation.Path;
+import jakarta.ws.rs.core.Response;
+
+import org.jboss.resteasy.reactive.RestResponse;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.tkit.quarkus.jpa.exceptions.ConstraintException;
+import org.tkit.quarkus.log.cdi.LogService;
+import org.tkit.quarkus.rs.mappers.OffsetDateTimeMapper;
+
+import gen.io.github.onecx.search.config.v1.model.ProblemDetailInvalidParamDTOV1;
+import gen.io.github.onecx.search.config.v1.model.ProblemDetailParamDTOV1;
+import gen.io.github.onecx.search.config.v1.model.ProblemDetailResponseDTOV1;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Mapper(uses = { OffsetDateTimeMapper.class })
+public abstract class ExceptionMapper {
+    @LogService(log = false)
+    public RestResponse<ProblemDetailResponseDTOV1> constraint(ConstraintViolationException ex) {
+        var dto = exception(TechnicalErrorKeys.CONSTRAINT_VIOLATIONS.name(), ex.getMessage());
+        dto.setInvalidParams(createErrorValidationResponse(ex.getConstraintViolations()));
+        return RestResponse.status(Response.Status.BAD_REQUEST, dto);
+    }
+
+    @LogService(log = false)
+    public RestResponse<ProblemDetailResponseDTOV1> exception(ConstraintException ce) {
+        var e = exception(ce.getMessageKey().name(), ce.getConstraints());
+        e.setParams(map(ce.namedParameters));
+        return RestResponse.status(Response.Status.BAD_REQUEST, e);
+    }
+
+    @LogService(log = false)
+    public RestResponse<ProblemDetailResponseDTOV1> optimisticLock(OptimisticLockException ex) {
+        var dto = exception(TechnicalErrorKeys.OPTIMISTIC_LOCK.name(), ex.getMessage());
+        return RestResponse.status(Response.Status.BAD_REQUEST, dto);
+    }
+
+    public List<ProblemDetailParamDTOV1> map(Map<String, Object> params) {
+        if (params == null) {
+            return List.of();
+        }
+        return params.entrySet().stream().map(e -> {
+            var item = new ProblemDetailParamDTOV1();
+            item.setKey(e.getKey());
+            if (e.getValue() != null) {
+                item.setValue(e.getValue().toString());
+            }
+            return item;
+        }).toList();
+    }
+
+    @Mapping(target = "invalidParams", ignore = true)
+    @Mapping(target = "removeInvalidParamsItem", ignore = true)
+    @Mapping(target = "removeParamsItem", ignore = true)
+    @Mapping(target = "params", ignore = true)
+    public abstract ProblemDetailResponseDTOV1 exception(String errorCode, String detail);
+
+    public abstract List<ProblemDetailInvalidParamDTOV1> createErrorValidationResponse(
```

Comment:
Move this to ExceptionMapper

Reply:
I dont understand, is this not the ExceptionMapper?